### PR TITLE
Kafka fixes and DStream.count fix for branch 0.7

### PR DESCRIPTION
**Pull request by tdas**
This pull request gets all the commits submitted by Sean Mc Namara on the streaming branch to branch-0.7.

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -173,6 +173,9 @@ object SparkBuild extends Build {
 
   def streamingSettings = sharedSettings ++ Seq(
     name := "spark-streaming",
+    resolvers ++= Seq(
+      "Akka Repository" at "http://repo.akka.io/releases/"
```
Comment (tdas) [inline project/SparkBuild.scala:177]: @mateiz 
I am not sure whether this is still necessary or not. I doubt it. I can remove this if you think the same.

**Comment by mateiz** [comment]
Jenkins, this is ok to test

**Comment by mateiz** [comment]
Thanks TD! Just want to make sure it passes the tests. The Akka repo is probably not needed because we added it in sharedSettings -- we can remove it later.

**Comment by mateiz** [comment]
It seems that the SBT on AMPLab Jenkins just hangs occasionally, but the tests pass for me, so I'll merge this. Thanks TD and Sean!

**Comment by mcnamaras** [comment]
My pleasure!  Thanks guys!

**Pull request merged**